lib/Controller/APIv2Controller.php

```diff
@@ -373,14 +373,16 @@ protected function getPreview(string $owner, int $fileId, string $filePath): arr
 		} else {
 			$this->view->chroot('/' . $owner . '/files');
 			$fileInfo = $this->view->getFileInfo($info['path']);
-			if (!$fileInfo instanceof FileInfo) {
+			if (!($fileInfo instanceof FileInfo)) {
 				$preview = $this->getPreviewFromPath($fileId, $filePath, $info);
 			} elseif ($this->preview->isAvailable($fileInfo)) {
-				$preview['source'] = $this->urlGenerator->linkToRouteAbsolute('core.Preview.getPreview', [
-					'file' => $info['path'],
-					'c' => $this->view->getETag($info['path']),
-					'x' => 150,
-					'y' => 150,
+				$preview['source'] = $this->urlGenerator->linkToRouteAbsolute('core.Preview.getPreviewByFileId', [
+					'fileId' => $fileId,
+					'c' => $fileInfo->getEtag(),
+					'x' => 250,
+					'y' => 250,
+					'forceIcon' => 0,
+					'a' => 0,
 				]);
```

Comment:
https://nextcloud24.local/index.php/core/preview?fileId=690&c=de6528a27e95f41c3ada4606f337b983&x=250&y=250&forceIcon=0&a=0
https://nextcloud24.local/index.php/core/preview?c=de6528a27e95f41c3ada4606f337b983&forceIcon=0&a=0&x=250&y=250&fileId=690

URLs still don't match for me so browser cache will not hit?

Reply:
Interesting the order is not the same as I'm observing locally:

activity: https://nextcloud.local/index.php/core/preview?forceIcon=0&a=0&x=250&y=250&fileId=5382&c=6217b04be4cce

file list (without https://github.com/nextcloud/server/pull/31346): /index.php/core/preview?fileId=5351&c=6217b04986bda&x=250&y=250&forceIcon=0&a=0

file list (with https://github.com/nextcloud/server/pull/31346): https://nextcloud.local/index.php/core/preview?forceIcon=0&a=0&x=250&y=250&fileId=5351&c=6217b04986bda

I wonder if it depends on the PHP version. I use PHP 7.4.

Reply:
Ok I found out the reasons, we were doing a asort($parameters) in server + some caching in redis. It should be good now with the latest version of https://github.com/nextcloud/server/pull/31346